code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm

```diff
@@ -163,14 +163,7 @@
 		return FALSE
 
 	if(scrubbing & SCRUBBING)
-		var/should_we_scrub = FALSE
-		for(var/id in env_gases)
-			if(id == /datum/gas/nitrogen || id == /datum/gas/oxygen)
-				continue
-			if(env_gases[id][MOLES] && filter_types[id])
```

Comment:
this is the line that broke it; `filter_types` isn't actually an associative list so `filter_types[id]` was null even if `id in filter_types` were true